src/main/kotlin/com/hibob/academy/filter/AuthenticationFilter.kt

```diff
@@ -0,0 +1,39 @@
+package com.hibob.academy.filter
+
+import com.hibob.academy.service.SessionService.Companion.SECRET_KEY
+import io.jsonwebtoken.Claims
+import io.jsonwebtoken.Jws
+import io.jsonwebtoken.Jwts
+import jakarta.ws.rs.container.ContainerRequestFilter
+import jakarta.ws.rs.container.ContainerRequestContext
+import jakarta.ws.rs.core.Response
+import jakarta.ws.rs.ext.Provider
+import org.springframework.stereotype.Component
+import org.springframework.web.bind.annotation.RequestMethod
+
+@Component // todo ask - who does init it actually?
+@Provider // todo ask - web return Boolean to authentication caller check?
+class AuthenticationFilter: ContainerRequestFilter {
+    override fun filter(requestContext: ContainerRequestContext) {
+
+        //verification logic
+        if (requestContext.uriInfo.path == "createToken") return
+
+        val res = verify(requestContext.cookies["JWT"]?.value)
```

Comment:
no need to be cheap with names, they don't cost money 😉 
please make the cookie name a const and reuse it in the resource class and here

Reply:
changed name, added const, used it at those files